### PR TITLE
Allow results page customization

### LINKED ISSUE #31: [4H] Allow results page customization
In `"metadata"`, we should provide ways to:

- [x] Change generated page title
- [x] Change generated page description
- [x] Change limit for `≈` sign (currently `±1%`)
- [x] Change the main repository URL (for commit references). This could be set in the GitHub action settings instead.

### PR DESCRIPTION
**Pull request by RemiBardon**
Closes #31.

### CODE REVIEW

File: src/action.py [hunk 1]
```diff
@@ -11,6 +11,8 @@ def main():
     parser.add_argument('hashref', help='Hash value of commit to compare against (reference commit)')
     parser.add_argument('hashcomp', help='Hash value of commit to compare')
     parser.add_argument('output_path', help='Path of folder to output the generated page')
+    parser.add_argument('repository_url', help='URL of the tested code repository (for commit hyperlinks)')
```
Comment (FlorianDussable) [inline src/action.py:14]: There is no sense in appending a local commit hash to another repo URL

File: src/main.py [hunk 2]
```diff
@@ -12,11 +12,16 @@ def main():
     parser.add_argument('ref_file_path', help='Path to JSON file to compare against (reference file)')
     parser.add_argument('comp_file_path', help='Path to JSON file to compare')
     parser.add_argument('output_path', help='Path of folder to output the generated page')
+    parser.add_argument('repository_url', help='URL of the tested code repository (for commit hyperlinks)')
```
Comment (FlorianDussable) [inline src/main.py:15]: Same

File: src/models.py [hunk 3]
```diff
@@ -23,8 +23,21 @@ def __init__(self, file_path: str, exit_diff: Diff, diffs: List[Diff]):
         self.exit_diff = exit_diff
         self.diffs = diffs
 class Metadata:
```
Comment (FlorianDussable) [inline src/models.py]: Not sure about adding external data to a model that's supposed to match the JSON structure

File: tests/test_main.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+from main import inner_main
+
+def test_the_program_runs():
+    inner_main(
+        ref_file_path='./tests/data/input/optimization1.out.json',
+        comp_file_path='./tests/data/input/optimization2.out.json',
+        output_path='./tests/.out/',
+        repository_url='https://github.com/chocoteam/choco-solver',
+    )
+
```
Comment (FlorianDussable) [inline tests/test_main.py:12]: I'd add assertions on file existence and on file size > 0

File: tests/test_main.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    file_content = f.read()
+
+def test_second_file_metadata_used_for_page_generation():
+    inner_main(
+        ref_file_path='./tests/data/input/legacy_optimization.out.json',
+        comp_file_path='./tests/data/input/new_optimization.out.json',
+        output_path='./tests/.out/',
+        repository_url='https://github.com/chocoteam/choco-solver',
+    )
+
```
Comment (FlorianDussable) [inline tests/test_main.py:23]: Same

File: tests/test_pagegen.py [hunk 6]
```diff
@@ -158,3 +173,33 @@ def test_show_both_commits_in_description():
...
+            TestResult(
+                '/home/evaluation/evaluation/pub/bench/XCSP18/CrosswordDesign/CrosswordDesign-03-4-rom_c18',
+                Diff('Exit value 1', 50, 40, -10, -20),
+                []
+            ),
+        ],
+    )
+
+    generate_page(tests_output_file_name, commit1, commit2, output_file_folder, input_data)
+    f = open('./tests/.out/test_metadata_are_used_to_generate_front_matter.md', 'r')
```
Comment (FlorianDussable) [inline tests/test_pagegen.py:212]: Same (assertions on file existence and on file size > 0)

**Review by FlorianDussable — CHANGES_REQUESTED** [review]

File: src/action.py (hunk 1 above)
Comment (RemiBardon) [reply src/action.py:14]: 100% true, I'll add it to the newly created `TestOutputMetadata` `class`.

File: src/models.py (hunk 3 above)
Comment (RemiBardon) [reply src/models.py]: I [split the `Metadata` model in `TestOutputMetadata` and `PageGenSettings`](https://github.com/chocoteam/PerformanceAction/pull/34/commits/16fb3616da449ed97fbb87231cdd59497aa55952).

File: tests/test_main.py (hunk 4 above)
Comment (RemiBardon) [reply tests/test_main.py:12]: [Done](https://github.com/chocoteam/PerformanceAction/pull/34/commits/61c3ebf7c1c86693c42ff639942c725548d1f4d2)

File: tests/test_pagegen.py (hunk 6 above)
Comment (RemiBardon) [reply tests/test_pagegen.py:212]: I use the file content here, doesn't apply

File: src/action.py (hunk 1 above)
Comment (RemiBardon) [reply src/action.py:14]: I propose this goes in a new issue. I created #35 for it.

**Review by morvillers — APPROVED** [review]
Looks good to me once the requested changes will be done

**Review by FlorianDussable — APPROVED** [review]
Seems good to me

**Pull request merged**